builtin/union_dataset.cc

```diff
@@ -0,0 +1,381 @@
+        vector<RowHash> result;
+        return result;
+    }
+
+    virtual bool knownRow(const Path & rowName) const
+    {
+        for (const auto & curr: getRowNames()) {
+            if (rowName == curr) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    virtual bool knownRowHash(const RowHash & rowHash) const
+    {
+        for (const auto & curr: getRowNames()) {
+            if (rowHash == curr.hash()) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    virtual RowName getRowName(const RowHash & rowHash) const
+    {
+        for (const auto & curr: getRowNames()) {
+            if (rowHash == curr.hash()) {
+                return curr;
+            }
+        }
+        throw ML::Exception("Row not known");
+    }
+
+    virtual MatrixNamedRow getRow(const RowName & rowName) const
+    {
+        string name = rowName.toUtf8String().rawString();
+        for (int i = 0; i < datasets.size(); ++i) {
+            string prefix;
+            string suffix;
+            tie(prefix, suffix) = getStrFix(i, datasets.size());
+            if (name.find(prefix) == 0 && name.rfind(suffix) == name.size() - suffix.size()) {
+                string subRowName = name.substr(prefix.size(), name.size() - prefix.size() - suffix.size());
+                MatrixNamedRow result =
+                    datasets[i]->getMatrixView()->getRow(Path(subRowName));
+                result.rowName = rowName;
+                //result.rowHash = ??; //TODO
+                return result;
+            }
+        }
+        MatrixNamedRow result;
+        return result;
+    }
+
+    virtual bool knownColumn(const Path & column) const
+    {
+        for (const auto & curr: getColumnNames()) {
+            if (curr == column) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    virtual ColumnName getColumnName(ColumnHash columnHash) const
+    {
+        for (const auto & curr: getColumnNames()) {
+            if (columnHash == curr.hash()) {
+                return curr;
+            }
+        }
+        throw ML::Exception("Column not known");
+    }
+
+    /** Return a list of all columns. */
+    virtual vector<ColumnName> getColumnNames() const
+    {
+        std::set<ColumnName> preResult;
+        for (const auto & d: datasets) {
+            auto columnNames = d->getColumnNames();
+            preResult.insert(columnNames.begin(), columnNames.end());
+        }
+        return vector<ColumnName>(preResult.begin(), preResult.end());
+    }
+
+    virtual MatrixColumn getColumn(const ColumnName & columnName) const
+    {
+        MatrixColumn result;
+        result.columnName = columnName;
+        //result.columnHash = columnName.hash(); //TODO
+        vector<std::tuple<RowName, CellValue> > res;
+        for (int i = 0; i < datasets.size(); ++i) {
+            const auto & d = datasets[i];
+            string prefix;
```

Comment:
This is going to be terribly slow unless we avoid the need to copy and manipulate strings